Update backport, release and sync documentation

flip1995 opened the pull request:
This updates the documentation after #13694. It is not based on that PR chain and can be merged independently, but should be merged after that PR.

This is partly pulled from #12762, but removing the Josh parts.

This includes instructions on how to publish `clippy_utils`.

Closes https://github.com/rust-lang/rust-clippy/issues/13556 (yes, this is the final PR :slightly_smiling_face:)

r? @blyxyas 

changelog: `clippy_utils` is now published to crates.io

BD103 commented on book/src/development/infrastructure/release.md:
```diff
@@ -7,112 +7,111 @@ Clippy is released together with stable Rust releases. The dates for these
-
-> _NOTE:_ This document is for stable Rust releases, not for point releases. For
-> point releases, step 1. and 2. should be enough.
+1. [Defining Remotes](#defining-remotes)
+1. [Bump Version](#bump-version)
+1. [Find the Clippy commit](#find-the-clippy-commit)
+1. [Update the `beta` branch](#update-the-beta-branch)
+1. [Update the `stable` branch](#update-the-stable-branch)
+1. [Tag the stable commit](#tag-the-stable-commit)
+1. [Update `CHANGELOG.md`](#update-changelogmd)
```
```suggestion
1. [Defining Remotes](#defining-remotes)
2. [Bump Version](#bump-version)
3. [Find the Clippy commit](#find-the-clippy-commit)
4. [Update the `beta` branch](#update-the-beta-branch)
5. [Update the `stable` branch](#update-the-stable-branch)
6. [Tag the stable commit](#tag-the-stable-commit)
7. [Update `CHANGELOG.md`](#update-changelogmd)
```

flip1995 replied:
I went with that, so when inserting a new chapter in the future, you don't have to update all the following numbers. But happy to change that.

BD103 reviewed (commented):
Just one small thing that caught my eye :)

blyxyas commented on book/src/development/infrastructure/release.md:
```diff
@@ -7,112 +7,111 @@ Clippy is released together with stable Rust releases. The dates for these
+Clippy repo.
 
-After finding the Clippy commit, it can be tagged with the release number.
+> Note: Only push the tag, once the Deploy GitHub action of the `beta` branch is
```
Ultra nit: This usually doesn't have a comma.

```suggestion
> Note: Only push the tag once the Deploy GitHub action of the `beta` branch is
```

blyxyas commented on book/src/development/infrastructure/sync.md:
Is there a reason to change the branch names? :thinking: I think those were fine.

flip1995 replied:
Yes. rustbot checks all PRs for merge commits and will comment on it, if it finds one. This can be opted out with allowing some PR titles to have merge commits:

https://github.com/rust-lang/rust-clippy/blob/8298da72e7b81fa30c515631b40fc4c0845948cb/triagebot.toml#L12-L15

In the Clippy repo it is `Rustup`, in the Rust repo it is `subtree update`. 

By using `rustup` as a branch name in the Clippy repo and `clippy-subtree-update` in the Rust repo, GitHub will automatically set the title when opening the PR to `Rustup` and `Clippy subtree update` respectively.

blyxyas reviewed (changes requested):
Good PR! :cowboy_hat_face: I had comments, some are more important than others.

flip1995 commented:
@rustbot ready

I addressed all comments in 9858a4e, except for https://github.com/rust-lang/rust-clippy/pull/13700#discussion_r1851083554. LMK, if you want me to change that. 

I'll rebase+squash once the PR is approved.

blyxyas reviewed (approved):
LGTM, thanks! :heart: :rainbow:
_Yeehaw_ :cowboy_hat_face:

blyxyas commented:
@rustbot author
